False positive W0613 with keyword only argument

### Bug description

Using a keyword-only argument is triggering a W0613(unused-argument) warning where it is not occurring if the (unused) argument is a keyword argument. Below, `B.add` does not issue the warning, but `D.add` does.

kwoa.py
```python
# pylint: disable=C*,R*
class A:
    def add(self, a, b=0):
        return a + b

class B(A):
    def add(self, a, b=0):
        return None

class C:
    def add(self, a, *, b=0):
        return a + b

class D(C):
    def add(self, a, *, b=0):
        return None
```

### Configuration

_No response_

### Command used

```shell
pylint kwoa.py
```


### Pylint output

```shell
************* Module kwoa
kwoa.py:16:24: W0613: Unused argument 'b' (unused-argument)
```


### Expected behavior

I believe there are two possibilities here but I had been operating under the assumption that this warning would not occur for `D.add` given that is not the behavior for `B.add`. The other option is that a warning should also be issued `B.add`.

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.0
Python 3.9.5 (default, May 18 2021, 12:31:01) 
[Clang 10.0.0 ]
```


### OS / Environment

_No response_

### Additional dependencies

_No response_

--- a/ChangeLog
+++ b/ChangeLog
@@ -154,6 +154,12 @@ Release date: TBA
   Closes #5408
   Ref PyCQA/astroid#1392
 
+* Fixed false positive for ``unused-argument`` when a method overridden in a subclass
+  does nothing with the value of a keyword-only argument.
+
+  Closes #5771
+  Ref PyCQA/astroid#1382
+
 * The issue template for crashes is now created for crashes which were previously not covered
   by this mechanism.
 

--- a/doc/whatsnew/2.13.rst
+++ b/doc/whatsnew/2.13.rst
@@ -247,6 +247,12 @@ Other Changes
   Closes #5408
   Ref PyCQA/astroid#1392
 
+* Fixed false positive for ``unused-argument`` when a method overridden in a subclass
+  does nothing with the value of a keyword-only argument.
+
+  Closes #5771
+  Ref PyCQA/astroid#1382
+
 * Fix ``unnecessary_dict_index_lookup`` false positive when deleting a dictionary's entry.
 
   Closes #4716

--- a/pylint/checkers/variables.py
+++ b/pylint/checkers/variables.py
@@ -2310,9 +2310,7 @@ class VariablesChecker(BaseChecker):
             if global_names and _import_name_is_global(stmt, global_names):
                 return
 
-        argnames = list(
-            itertools.chain(node.argnames(), [arg.name for arg in node.args.kwonlyargs])
-        )
+        argnames = node.argnames()
         # Care about functions with unknown argument (builtins)
         if name in argnames:
             self._check_unused_arguments(name, node, stmt, argnames)

--- a/tests/functional/u/unused/unused_argument.py
+++ b/tests/functional/u/unused/unused_argument.py
@@ -87,3 +87,16 @@ class BBBB(object):
     def __init__(self, arg):  # [unused-argument]
         """Constructor with an extra parameter. Should raise a warning"""
         self.spam = 1
+
+
+class Ancestor:
+    def __init__(self):
+        self.thing = None
+
+    def set_thing(self, thing, *, other=None):  # [unused-argument]
+        self.thing = thing
+
+class Descendant(Ancestor):
+    def set_thing(self, thing, *, other=None):
+        """Subclass does not raise unused-argument"""
+        self.thing = thing

--- a/tests/functional/u/unused/unused_argument.py
+++ b/tests/functional/u/unused/unused_argument.py
@@ -89,6 +89,8 @@ class BBBB(object):
         self.spam = 1
 
 
+# Regression test for https://github.com/PyCQA/pylint/issues/5771
+# involving keyword-only arguments
 class Ancestor:
     def __init__(self):
         self.thing = None

--- a/tests/functional/u/unused/unused_argument.txt
+++ b/tests/functional/u/unused/unused_argument.txt
@@ -6,3 +6,4 @@ unused-argument:61:21:61:24:AAAA.method:Unused argument 'arg':INFERENCE
 unused-argument:68:0:None:None:AAAA.selected:Unused argument 'args':INFERENCE
 unused-argument:68:0:None:None:AAAA.selected:Unused argument 'kwargs':INFERENCE
 unused-argument:87:23:87:26:BBBB.__init__:Unused argument 'arg':INFERENCE
+unused-argument:96:34:96:39:Ancestor.set_thing:Unused argument 'other':INFERENCE

--- a/tests/functional/u/unused/unused_argument.txt
+++ b/tests/functional/u/unused/unused_argument.txt
@@ -6,4 +6,4 @@ unused-argument:61:21:61:24:AAAA.method:Unused argument 'arg':INFERENCE
 unused-argument:68:0:None:None:AAAA.selected:Unused argument 'args':INFERENCE
 unused-argument:68:0:None:None:AAAA.selected:Unused argument 'kwargs':INFERENCE
 unused-argument:87:23:87:26:BBBB.__init__:Unused argument 'arg':INFERENCE
-unused-argument:96:34:96:39:Ancestor.set_thing:Unused argument 'other':INFERENCE
+unused-argument:98:34:98:39:Ancestor.set_thing:Unused argument 'other':INFERENCE

